### PR TITLE
Fixed Makefile dependencies, one test for request handlers

### PR DESCRIPTION
**Pull request by natalieethell**

### CODE REVIEW

**Comment by natalieethell** [comment]
status handler still needs to output existing handlers

File: Makefile [hunk 1]
```diff
@@ -27,24 +27,29 @@ config_parser_test:
...
+	$(CC) $(CFLAGS) -std=c++0x -isystem ${GTEST_DIR}/include -I${SRC_DIR} $(TEST_DIR)/response_test.cc $(SRC_DIR)/status_count.cc $(SRC_DIR)/response.cc ${GTEST_DIR}/src/gtest_main.cc libgtest.a $(LIBFLAGS) -o response_test
+
+request_test:
+	$(CC) $(CFLAGS) -std=c++0x -isystem ${GTEST_DIR}/include -I${SRC_DIR} $(TEST_DIR)/request_test.cc $(SRC_DIR)/server_config.cc $(SRC_DIR)/config_parser.cc $(SRC_DIR)/status_count.cc $(SRC_DIR)/request.cc $(SRC_DIR)/request_handler.cc $(SRC_DIR)/response.cc ${GTEST_DIR}/src/gtest_main.cc libgtest.a $(LIBFLAGS) -o request_test
+
+status_count_test:
+	$(CC) $(CFLAGS) -std=c++0x -isystem ${GTEST_DIR}/include -I${SRC_DIR} $(TEST_DIR)/status_count_test.cc $(SRC_DIR)/server_config.cc $(SRC_DIR)/status_count.cc $(SRC_DIR)/response.cc $(SRC_DIR)/request.cc $(SRC_DIR)/config_parser.cc $(SRC_DIR)/request_handler.cc ${GTEST_DIR}/src/gtest_main.cc libgtest.a $(LIBFLAGS) -o status_count_test
 
 # NOTE: config_parser_test must be executed first -- the command does the googletest setup neccessary for the later tests
-test: config_parser_test server_config_test response_test
```
Comment (binerys) [inline Makefile]: All these tests shouldn't be removed

File: Makefile [hunk 2]
```diff
@@ -27,24 +27,29 @@ config_parser_test:
...
+test: config_parser_test request_test
 	./config_parser_test
-	./server_config_test
-	./response_test
```
Comment (binerys) [inline Makefile]: don't remove these!

File: src/request_handler.cc [hunk 3]
```diff
@@ -93,4 +100,60 @@ RequestHandler::Status StaticHandler::HandleRequest(const Request& request, Resp
...
+
+std::string StaticHandler::GetName() {
+  return "StaticHandler";
+}
+
+/* STATUS HANDLER */
+
+RequestHandler::Status StatusHandler::Init(const std::string& uri_prefix, const NginxConfig& config){
+  uri_ = uri_prefix;
+  std::cout << "Called status handler Init" << std::endl;
```
Comment (binerys) [inline src/request_handler.cc]: Use Boost logging (BOOST_LOG_TRIVIAL) not cout

File: src/request_handler.cc [hunk 4]
```diff
@@ -93,4 +100,60 @@ RequestHandler::Status StaticHandler::HandleRequest(const Request& request, Resp
...
+  return RequestHandler::Status::OK;
+}
+
+RequestHandler::Status StatusHandler::HandleRequest(const Request& request, Response* response) {
+
+  std::cout << "Called status handler handle request" << std::endl;
+  
+  // // TODO: finish implementing
```
Comment (binerys) [inline src/request_handler.cc]: Remove this comment if it's outdated

File: src/server.cc [hunk 5]
```diff
@@ -1,5 +1,8 @@
+#include "request_handler.h"
+#include "status_count.h"
 #include "server.h"
 #include "http_constants.h"
+#include <vector>
```
Comment (binerys) [inline src/server.cc]: Put #includes (besides server.h and http_constants.h) in server.h. It should work (it's worked for me lol

File: test/request_handler_test.cc [hunk 6]
```diff
@@ -4,12 +4,37 @@
 #include <boost/algorithm/string.hpp>
```
Comment (binerys) [inline test/request_handler_test.cc]: unnecessary includes

File: test/request_handler_test.cc [hunk 7]
```diff
@@ -4,12 +4,37 @@
...
 #include <boost/filesystem.hpp>
```
Comment (binerys) [inline test/request_handler_test.cc]: unneccessary includes

**Comment by binerys** [comment]
Can you also delete all the commented out code in request_handler_test.h?

File: src/request_handler.cc (hunk 3 above)
Comment (natalieethell) [inline src/request_handler.cc]: gotcha thanks

File: Makefile (hunk 2 above)
Comment (natalieethell) [inline Makefile]: cool yeah I'll put em back when I'm done testing specific files

**Pull request merged**